### DISCUSSION TITLE
default encoding / supported encodings

### DISCUSSION THREAD

**Discussion by massimosala** [micropython/micropython#11955] [General]
Hi

I read the docs, also

> MicroPython differences from CPython
> https://docs.micropython.org/en/latest/genrst/index.html

but I cannot find an explanation.

Please see the following tests.

**Cpython 3**

```
>>> ord('à')
224

>>> 'à'.encode("ascii")
UnicodeEncodeError: 'ascii' codec can't encode character '\xe0' in position 0: ordinal not in range(128)

>>> 'à'.encode("latin1")
b'\xe0'

>>> 'à'.encode("utf-8")
b'\xc3\xa0'

>>> bytes("à", "latin1")
b'\xe0'

>>> bytes("à", "utf-8")
b'\xc3\xa0'
```

---

**Micropython (esp8266)**

```
>>> chr(224).encode()
b'\xc3\xa0'

>>> chr(224).encode("ascii")
b'\xc3\xa0'

>>> chr(224).encode("latin1")
b'\xc3\xa0'

>>> chr(224).encode("utf-8")
b'\xc3\xa0'

>>> bytes(chr(224), "ascii")
b'\xc3\xa0'

>>> bytes(chr(224), "latin1")
b'\xc3\xa0'

>>> bytes(chr(224), "utf-8")
b'\xc3\xa0'
```

Note: at the REPL, with Windows Teraterm, I cannot type accented letters. So I use chr().

It seems to me, at least on the esp8266, the only codec supported is utf-8.
Am I missing something?

**Comment by dlech** [comment]
You are not missing anything. MicroPython only supports UTF-8.

**Reply by peterhinch** [reply]
As documented [here](https://github.com/micropython/micropython/wiki/Differences#implementation-differences).

**Comment by mattytrentini** [comment]
Yes, as @dlech and @peterhinch have mentioned, the only supported encoding is UTF-8. 

Just to explain why you needed to use `chr`; the MicroPython _REPL doesn't support unicode_ - see #2789 for details. To clarify, the MicroPython _interpreter_ understands unicode just fine - but the _REPL_ currently processes data character-by-character which fails for unicode. You can use `chr` or you can use the paste buffer (Ctrl+E) at the REPL. The paste buffer works because the _whole_ text block is sent to the interpreter.

**Comment by massimosala** [comment]
Thanks to all
